python/ray/tests/test_runtime_env_plugin.py

```diff
@@ -72,6 +74,63 @@ def f():
         assert output == {"env_value": "42", "tmp_content": "hello", "nice": 19}
 
 
+MY_PLUGIN_FOR_HANG_CLASS_PATH = "ray.tests.test_runtime_env_plugin.MyPluginForHang"
+my_plugin_setup_times = 0
+
+
+# This plugin will hang when first setup, second setup will ok
+class MyPluginForHang(RuntimeEnvPlugin):
+    env_key = "MY_PLUGIN_FOR_HANG_TEST_ENVIRONMENT_KEY"
+
+    @staticmethod
+    def validate(runtime_env_dict: dict) -> str:
+        return "True"
+
+    @staticmethod
+    def create(uri: str, runtime_env: dict, ctx: RuntimeEnvContext) -> float:
+        global my_plugin_setup_times
+        my_plugin_setup_times += 1
+
+        # first setup
+        if my_plugin_setup_times == 1:
+            # sleep forever
+            sleep(3600)
+
+    @staticmethod
+    def modify_context(
+        uri: str, plugin_config_dict: dict, ctx: RuntimeEnvContext
+    ) -> None:
+        global my_plugin_setup_times
+        ctx.env_vars[MyPluginForHang.env_key] = str(my_plugin_setup_times)
+
+
+def test_plugin_hang(ray_start_regular):
+    @ray.remote(num_cpus=0.1)
```

Comment:
before this PR, this case will be failed.

Reply:
Nice! Thanks for adding this!